### PR TITLE
[Robot] Prerelease fixes for unstable tests

### PR DESCRIPTION
**Pull request by rjanjanam**
Fixes made to address the prerelease fixes under the robot unstable build

To test the fixes
Create a prelease org 
cci org scratch prerelease <orgname> --days <number>
cci flow run qa_org --org <orgname>
cci task run robot --org <orgname> -o suites robot/Cumulus/tests/browser  -o include unstable -o vars BROWSER:headlesschrome

Known failures: 
There are atleast 2 RD test failures because of the issues below
https://salesforce.quip.com/ItfxATFc2AJp

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 1]
```diff
@@ -379,7 +377,7 @@ def validate_current_and_next_year_values(self, amount, rdtype=None):
             if rdtype == "Open":
                 next_year_value = next_year_value + (12-next_year_count)*int(amount)
             values['Current Year Value']=f"${curr_year_value}.00"
-            values['Next Year Value']=f"${ next_year_value}.00"
+            #values['Next Year Value']=f"${ next_year_value}.00"
```
Comment (Safiyab) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:380]: Did u intentionally comment it out ?

File: robot/Cumulus/tests/browser/2_batch_data_imports/bdi_reprocess_failed_DI.robot [hunk 2]
```diff
@@ -8,7 +8,7 @@ Suite Setup     Run keywords
 ...             Open Test Browser
 ...             Setup Variables
 ...             Enable Advanced Mapping
-Suite Teardown  Capture Screenshot and Delete Records and Close Browser
+#Suite Teardown  Capture Screenshot and Delete Records and Close Browser
```
Comment (Safiyab) [inline robot/Cumulus/tests/browser/2_batch_data_imports/bdi_reprocess_failed_DI.robot:11]: I think u forgot to uncomment this

File: robot/Cumulus/resources/DataImportPageObject.py [hunk 3]
```diff
@@ -42,7 +43,8 @@ def click_close_button(self):
 
     def open_data_import_record(self,di_name):
         """Clicks on the specified data import record to open the record"""
-        self.pageobjects.current_page_should_be("Listing","DataImport__c")
+        #self.pageobjects.current_page_should_be("Listing","DataImport__c")
```
Comment (Safiyab) [inline robot/Cumulus/resources/DataImportPageObject.py:46]: Did u intentionally comment it out ?

**Review by Safiyab — APPROVED** [review]

File: robot/Cumulus/resources/RecurringDonationsPageObject.py (hunk 1 above)
Comment (rjanjanam) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py:380]: Yes this validation does not apply anymore

File: robot/Cumulus/resources/DataImportPageObject.py (hunk 3 above)
Comment (rjanjanam) [reply robot/Cumulus/resources/DataImportPageObject.py:46]: Thanks. Will fix it

File: robot/Cumulus/resources/OpportunityPageObject.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ def ensure_opportunity_details_are_loaded(self,objectID, value):
         self.npsp.navigate_to_and_validate_field_value("Opportunity Name", "contains", value)
 
     def navigate_to_matching_gifts_page(self):
-        if self.npsp.latest_api_version == 50.0:
+        if self.npsp.latest_api_version == 51.0:
```
Comment (skristem) [inline robot/Cumulus/resources/OpportunityPageObject.py:27]: do u even need this line anymore? if its same logic for 50 and 51 I think we should just remove this if statement

File: robot/Cumulus/resources/RecurringDonationsPageObject.py (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py:380]: please remove this if not needed

File: robot/Cumulus/resources/locators_51.py [hunk 5]
```diff
@@ -183,7 +185,7 @@
         },
     'payments':{
         'date_loc':"//*[@id='pmtTable']/tbody/tr/td[3]/div//input",
-        'no_payments':'//tbody/tr[./th//a[contains(@title,"PMT")]]/td[3]',
+        'no_payments':'//tbody/tr[./th//a[contains(@title,"PMT")]]/td[2]',
```
Comment (skristem) [inline robot/Cumulus/resources/locators_51.py:188]: can we use something else here instead of indexing as seems like index changing

**Review by skristem — APPROVED** [review]

**Pull request merged**